### ISSUE TITLE
Segmentation fault when exporting BatchNorm with aot

### ISSUE THREAD

**Issue by Hacker1337** [iree-org/iree-turbine#693]
Every time I run aot.export with model containing batch normalization, python crashes with segmentation fault.

And this comes from iree-turbine level, torch-mlir works fine with these models.

Code to reproduce

```python
import torch
import iree.turbine.aot as aot

print(torch.__version__)


import subprocess

result = subprocess.run(
    "pip list | grep iree", shell=True, capture_output=True, text=True
)
print(result.stdout)


import sys
print(f"Python version: {sys.version}")


##########################
### model
##########################
class BatchNormNN(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.bn = torch.nn.BatchNorm1d(32)

    def forward(self, x):
        x = self.bn(x)
        return x


model = BatchNormNN()
input = (torch.randn(5, 32),)

# iree export

import iree.turbine.aot as aot

export_output: aot.ExportOutput = aot.export(model, *input)

export_output.save_mlir("iree_export.mlir")

```

My output
```
2.5.1+cpu
iree-base-compiler 3.3.0
iree-base-runtime  3.3.0
iree-turbine       3.3.0

Python version: 3.11.0 | packaged by conda-forge | (main, Jan 14 2023, 12:27:40) [GCC 11.3.0]
Segmentation fault (core dumped)
```

Also reproduces with nighties iree_turbine-3.4.0rc20250408

I checked similar issues, #539 and #349 do not seem to be the same.
